packages/graphql-language-service-server/src/GraphQLCache.ts

```diff
@@ -75,15 +76,21 @@ export class GraphQLCache implements GraphQLCacheInterface {
   _typeExtensionMap: Map<Uri, number>;
   _fragmentDefinitionsCache: Map<Uri, Map<string, FragmentInfo>>;
   _typeDefinitionsCache: Map<Uri, Map<string, ObjectTypeInfo>>;
+  _parser: typeof parseDocument;
 
-  constructor(configDir: Uri, graphQLConfig: GraphQLConfig) {
+  constructor(
+    configDir: Uri,
+    graphQLConfig: GraphQLConfig,
+    parser?: typeof parseDocument,
+  ) {
     this._configDir = configDir;
     this._graphQLConfig = graphQLConfig;
     this._graphQLFileListCache = new Map();
     this._schemaMap = new Map();
     this._fragmentDefinitionsCache = new Map();
     this._typeDefinitionsCache = new Map();
     this._typeExtensionMap = new Map();
+    this._parser = parser || parseDocument;
```

Comment:
we should probably only supply the default in one place here, and remove the || in `MessageProcessor`. Also i think we should use `??` here.

Reply:
Fixed!